packages/transaction_signing_gateway/lib/storage/cosmos_key_info_storage.dart

```diff
@@ -167,4 +167,28 @@ class CosmosKeyInfoStorage implements KeyInfoStorage {
     final privateCreds = await getPrivateCredentials(walletLookupKey);
     return right(privateCreds.isRight());
   }
+
+  @override
+  Future<Either<CredentialsStorageFailure, Unit>> updatePublicWalletInfo({required WalletPublicInfo info}) async {
+    try {
+      final publicInfoKey = _publicInfoKey(
+        chainId: info.chainId,
+        walletId: info.walletId,
+      );
+
+      return _plainDataStore.readPlainText(key: publicInfoKey).flatMap((walletInfo) async {
+        if (walletInfo == null) {
+          return left(const CredentialsStorageFailure("Wallet not found"));
+        }
+
+        final publicInfoJson = await compute(jsonEncode, WalletPublicInfoSerializer.toMap(info));
+        await _plainDataStore.savePlainText(key: publicInfoKey, value: publicInfoJson);
+
+        return right(unit);
```

Comment:
you're ignoring potential errors in `savePlainText` and always returning `right(unit)`. you should return result of `savePlainText` directly so that the resulting future will fail with `left(..)` if saving fails
```suggestion
        return _plainDataStore.savePlainText(key: publicInfoKey, value: publicInfoJson);
```

Reply:
Got it.